File: dotnet/System/Workspace/Allors.Workspace.Domain/Request/Between.cs
// <copyright file="Between.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Request
{
    using System.Collections.Generic;
    using Allors.Workspace.Meta;
    using Allors.Workspace.Request.Visitor;

    public class Between : IRolePredicate
    {
        public Between(IRoleType roleType = null) => this.RoleType = roleType;

        public IEnumerable<object> Values { get; set; }

        public IEnumerable<IRoleType> Paths { get; set; }

        public string Parameter { get; set; }

        public IRoleType RoleType { get; set; }

        public void Accept(IVisitor visitor) => visitor.VisitBetween(this);
    }
}

File: typescript/libs/workspace/system/domain/src/lib/data/between.ts
import { RoleType } from '@allors/workspace/system/meta';
import { IUnit } from '../types';
import { ParameterizablePredicateBase } from './parameterizable-predicate';

export interface Between extends ParameterizablePredicateBase {
  kind: 'Between';

  roleType: RoleType;

  values?: IUnit[];

  paths?: RoleType[];
}
